File: tests/fixtures/ssrf_toctou_safe.py
import os
import requests


def fetch_healthcheck():
    return requests.get("https://status.example.com/health", timeout=5)


def read_file(path):
    try:
        with open(path, encoding="utf-8") as source:
            return source.read()
    except FileNotFoundError:
        return ""


def create_exclusively(path):
    descriptor = os.open(path, os.O_CREAT | os.O_EXCL | os.O_WRONLY, 0o600)
    os.close(descriptor)

File: tests/fixtures/ssrf_toctou_safe.js
const fs = require("node:fs");
const https = require("node:https");

function fetchHealthcheck() {
  return https.get("https://status.example.com/health");
}

function readFile(path) {
  try {
    return fs.readFileSync(path, "utf8");
  } catch (error) {
    if (error.code === "ENOENT") return "";
    throw error;
  }
}

module.exports = { fetchHealthcheck, readFile };
